service/interfaces.go

```diff
@@ -51,3 +54,8 @@ type Responder interface {
 	StatusCode(http.ResponseWriter, int)
 	Bytes(context.Context, http.ResponseWriter, int, []byte)
 }
+
+type DatasetAPIClient interface {
+	GetDatasets(ctx context.Context, uToken, svcToken, collectionID string, params *dataset.QueryParams) (dataset.List, error)
+	Checker(ctx context.Context, state *healthcheck.CheckState) error
```

Comment:
I see the definition of the checker as part of the contract, but it not being used I think